MissingNumber.java

```diff
@@ -0,0 +1,27 @@
+// Time Complexity : O(n)
+// Space Complexity : O(n)
```

Comment:
Please verify the space complexity. 
Generally, we don't include the result space when we tell the space complexity.